commands/db_scrape.ts

```diff
@@ -46,6 +47,26 @@ const LOADABLE_EXTENSIONS = [".js", ".cjs", ".mjs", ".ts"];
 
 export type TaskHandle = Parameters<TaskCallback>[0];
 
+export function resolveDate(dateString: string): DateTime<true> {
+  const date = DateTime.fromISO(dateString);
+  return date.isValid ? date : DateTime.now();
+}
+
+export interface SourceResponse<T> {
+  data: T[];
+}
+
+export function isValidDataResponse<T>(
+  response: unknown,
+): response is SourceResponse<T> {
+  return (
+    typeof response === "object" &&
+    response !== null &&
+    "data" in response &&
+    Array.isArray(response.data)
+  );
+}
```

Comment:
honestly this validator function is completely inadequate for what it's supposed to do